### PR TITLE
Feat/edit service record

### PR DESCRIPTION
**Pull request by batch08**
- added edit service record
- applied date locale in FormikDatePicker component
- added county filter

### CODE REVIEW

File: packages/webapp/src/components/lists/ReportList/index.tsx [hunk 1]
```diff
@@ -882,6 +926,23 @@ const ReportList = memo(function ReportList({ title }: ReportListProps): JSX.Ele
 					return item?.address?.city
 				}
 			},
+			{
+				key: 'county',
+				headerClassName: styles.headerItemCell,
+				itemClassName: styles.itemCell,
+				name: t('customFilters.county'),
+				onRenderColumnHeader: function onRenderColumnHeader(key, name, index) {
```
Comment (darthtrevino) [inline packages/webapp/src/components/lists/ReportList/index.tsx:934]: can use object shorthand here:

```
name: <xx>
onRenderColumnHeader(key, name, index) {
}
```

File: packages/webapp/src/components/lists/ReportList/index.tsx [hunk 2]
```diff
@@ -323,6 +331,26 @@ const ReportList = memo(function ReportList({ title }: ReportListProps): JSX.Ele
 	// #endregion Report filter functions
 
 	// #region Service Report functions
+	const handleUpdateServiceAnswer = async (values) => {
```
Comment (darthtrevino) [inline packages/webapp/src/components/lists/ReportList/index.tsx:334]: This is fine, but as a minor point, `function`s are probably the right choice over lambas for functions of this size. They provide call-stack information via their function name, where lambdas are anonymous in call stacks. Personally I only use lambdas for tiny callbacks or when I'm being deliberate about binding the `this` context.

File: packages/api/src/interactors/UpdateServiceAnswerInteractor.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+	public constructor(localization: Localization, services: ServiceCollection) {
+		this.#localization = localization
+		this.#services = services
+	}
+
+	public async execute(serviceAnswer: ServiceAnswerInput): Promise<ServiceResponse> {
+		if (!serviceAnswer.serviceId) {
+			return {
+				service: null,
```
Comment (darthtrevino) [inline packages/api/src/interactors/UpdateServiceAnswerInteractor.ts:29]: We repeat a same-ish block here a few times, let's add a `failureWithMessage(message: string)` function at the module level that assembles this object

**Review by darthtrevino — APPROVED** [review]

File: packages/webapp/src/components/lists/ReportList/index.tsx (hunk 1 above)
Comment (batch08) [reply packages/webapp/src/components/lists/ReportList/index.tsx]: applied code review

File: packages/webapp/src/components/lists/ReportList/index.tsx (hunk 2 above)
Comment (batch08) [reply packages/webapp/src/components/lists/ReportList/index.tsx]: applied code review

File: packages/api/src/interactors/UpdateServiceAnswerInteractor.ts (hunk 3 above)
Comment (batch08) [reply packages/api/src/interactors/UpdateServiceAnswerInteractor.ts:29]: will apply in the next iteration

**Pull request merged**